src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -367,6 +373,11 @@ public ErrorDiagnostic CannotResolveFunctionOverload(IList<string> overloadSigna
                 "BCP066",
                 $"Function \"{functionName}\" is not valid at this location. It can only be used in resource declarations.");
 
+            public ErrorDiagnostic ObjectRequiredForMethodAccess(TypeSymbol wrongType) => new ErrorDiagnostic(

```

Comment:
Side comment: After we get wider usage (past 0.3), we should stop reusing old codes. People will have articles, SO questions, etc. that will get invalidated if we reuse a code for a new error. (We can probably add tests with banned codes to stop ourselves from doing it accidentally at that point.)

Reply:
Yeah totally. Would make sense to add a test to enforce this.

Reply:
Yeah, for sure.